src/api/browse.ts

```diff
@@ -8,8 +8,21 @@ import { FacetFilterModel } from "../models/facets";
 import { fetchJson } from "../utils/utils";
 import { showMessage } from "../components/messages/usage";
 
-const MASS_URL = process.env.REACT_APP_MASS_URL;
-const METLDATA_URL = process.env.REACT_APP_METLDATA_URL;
+/**
+ * Checks and cleans up URL if last character is or is not forward slash
+ * Duplicate of utils version because of infinite recursion error, needs refactoring
+ * @param url - String of URL
+ * @param endSlash - whether we want to have an end slash or not
+ * @returns URL as specified
+ */
+const cleanUrl = (url: string, endSlash : boolean = true) => {
+  const lastCharSlash : boolean = url.slice(-1) === "/"
+  return endSlash ? lastCharSlash ? url : url + "/" : lastCharSlash ? url.slice(0,-1) : url
+}
+
+const CLIENT_URL : URL = new URL(cleanUrl(process.env.REACT_APP_CLIENT_URL as string))
+const MASS_URL : URL = new URL(cleanUrl(process.env.REACT_APP_MASS_URL as string), CLIENT_URL);
+const METLDATA_URL : URL = new URL(cleanUrl(process.env.REACT_APP_METLDATA_URL as string), CLIENT_URL)
```

Comment:
```suggestion
const CLIENT_URL : URL = new URL(cleanUrl(process.env.REACT_APP_CLIENT_URL!))
const MASS_URL : URL = new URL(cleanUrl(process.env.REACT_APP_MASS_URL!), CLIENT_URL);
const METLDATA_URL : URL = new URL(cleanUrl(process.env.REACT_APP_METLDATA_URL!), CLIENT_URL)
```

A bit shorter and expresses better what we are assuring here, namely that these vars have defined values.

Reply:
Done